packages/emails/email-manager.ts

```diff
@@ -27,45 +27,32 @@ import OrganizerScheduledEmail from "./templates/organizer-scheduled-email";
 import type { TeamInvite } from "./templates/team-invite-email";
 import TeamInviteEmail from "./templates/team-invite-email";
 
+// eslint-disable-next-line
+const sendEmail = (prepare: () => any) => {
+  return new Promise((resolve, reject) => {
+    try {
+      const email = prepare();
+      resolve(email.sendEmail());
+    } catch (e) {
+      reject(console.error(`${prepare.constructor.name}.sendEmail failed`, e));
+    }
+  });
+};
+
```

Comment:
```suggestion
interface EmailPreparation<T extends BaseEmail> {
  new (): T;
}

function sendEmail<T extends BaseEmail>(prepare: () => EmailPreparation<T>): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    try {
      const EmailClass = prepare();
      const email = new EmailClass();
      email.sendEmail();
      resolve();
    } catch (e) {
      reject(console.error(`${prepare.constructor.name}.sendEmail failed`, e));
    }
  });
}
```

I manage to make some improvements trying to type this correctly but didnt get much further than this. 

@zomars can your TS wizardry help me out here

Reply:
I actually managed to get this to work... Not sure if its the prettiest solution there is.
<img width="844" alt="CleanShot 2023-02-22 at 03 56 02@2x" src="https://user-images.githubusercontent.com/55134778/220517941-25870058-6a1e-4f72-a142-31bcabdcc637.png">

Reply:
Thanks for finding this @sean-brydon. I'll wait and see what @zomars thinks before merging.

Reply:
Maybe this is acceptable https://github.com/calcom/cal.com/pull/7270/files#r1115417367